chainer_compiler/elichika/parser/vevaluator.py

```diff
@@ -1173,105 +1213,185 @@ def veval_ast_for(astc : 'AstContext', local_field : 'values.Field', graph : 'Gr
 
     return None
 
-def veval_ast_continue(astc : 'AstContext', local_field : 'values.Field', graph : 'Graph'):
+def veval_ast_continue(astc : 'AstContext', local_field : 'values.Field', graph : 'Graph', option : 'VEvalOption' = None):
     assert(isinstance(astc.nast, gast.gast.Continue))
     return None
 
-def veval_ast_break(astc : 'AstContext', local_field : 'values.Field', graph : 'Graph'):
+def veval_ast_break(astc : 'AstContext', local_field : 'values.Field', graph : 'Graph', option : 'VEvalOption' = None):
     assert(isinstance(astc.nast, gast.gast.Break))
     return None
 
+def veval_ast_with(astc : 'AstContext', local_field : 'values.Field', graph : 'Graph', option : 'VEvalOption' = None):
```

Comment:
(TODO for myself: understand this)

Reply:
A description of how I designed the support for `With` syntax with flags.
- veval_ast(astc<gast.gast.withitem>, ...) behaves almost like `veval_ast_assign` assigning the output of processed `astc.nast.context_expr` to attribute `astc.nast.optional_vars`. Before assigning the output of `astc.nast.context_expr` however, we process  `__enter__` function of the context_expr attribute and use the output in place of it. This handles the case where the custom __enter__ function doesn't return self object.

- Compiler flags such as `with flags.for_unroll()` is handled by adding their function names to `builtin_function_converters`, with endpoint `VEvalOptionFunction`. This class extends `FunctionBase` and it's `vcall` just sets the corresponding flag of VEvalOption object option (through a shared `vevaluator.common_flags_cache`). (Note: I can probably remove this cache and modify the option object directly since vcall has access to it after the interface change I did later.)

-  veval_ast(astc<gast.gast.with>, ...) processes the body after all the withitems have been processed and then calls the withitems' context_expr' `__exit__` functions. The purpose of `ExitStack` is to allow multiple `with option.<flag>()` succintly (L1241). The option flags are set only in the with scope(In this case `ExitStack` scope) as they are unset in `__exit__` call of VEvalOption, thanks to `auto_set_unset`.

Reply:
Thanks for you explanation!

> I can probably remove this cache and modify the option object directly since vcall has access to it after the interface change I did later.

This sounds great